### ISSUE TITLE
Streaming connections don't work with a header.

### ISSUE THREAD

**Issue by skorokithakis** [mholt/caddy#1253]
I'm running Caddy and having it proxy a Gweet (https://github.com/skorokithakis/gweet/releases) instance.

My Caddyfile is:

~~~
http://localhost/ {
    header / Strict-Transport-Security "max-age=31536000; includeSubdomains";
    proxy / http://localhost:1234/ {
        transparent
    }
}
~~~

You can just launch Gweet and Caddy with that config, connect with:

curl --raw "http://localhost/stream/test/?streaming=1"

and create a message with:

curl -X POST "http://localhost/stream/test/?foo=bar"

Normally, you should see the message appear in the stream, but after 0.9.0 it doesn't any more.

If you remove the header, everything works properly.

**Comment by mholt** [comment]
When you say "doesn't work" what's the error you get? (Hint: is it "not a Flusher"? -- I have a hypothesis. Take out the header directive and try again.)

**Comment by skorokithakis** [comment]
Sorry, I should have been more descriptive. No error, it connects and waits forever, but no data streams in. Yes, t is "no flusher". It works without the header, as I said above.

**Comment by mholt** [comment]
Oh, oops. I missed that part, okay, I think this has been fixed on master in 14500d8204c806edc69daa1725df7c68aa6f6347 -- if you have the ability to build from source, give it a shot and let me know! If not, feel free to close this.

**Comment by mholt** [comment]
(Assuming this is already fixed)
